client/angular/controllers/AdminController.js

```diff
@@ -2,20 +2,24 @@
 
     angular.module("TwitterWallApp").controller("AdminController", AdminController);
 
-    AdminController.$inject = ["$scope", "adminDashDataService", "$sce", "tweetTextManipulationService"];
+    AdminController.$inject = ["$scope", "adminDashDataService", "$sce", "tweetTextManipulationService", "$routeParams"];
 
-    function AdminController($scope, adminDashDataService, $sce, tweetTextManipulationService) {
+    function AdminController($scope, adminDashDataService, $sce, tweetTextManipulationService, $routeParams) {
         $scope.loggedIn = false;
         $scope.ctrl = {};
         $scope.tweets = [];
         $scope.motd = "";
+        $scope.errorMessage = "";
 
         $scope.deleteTweet = adminDashDataService.deleteTweet;
 
         adminDashDataService.authenticate().then(function () {
             $scope.loggedIn = true;
         }, function () {
             adminDashDataService.getAuthUri().then(function (uri) {
+                if ($routeParams.status) {
```

Comment:
The `status` parameter should probably be matched against "unauthorized" here, since otherwise it's effectively acting as a 404 page (even if the user is authorized).